File: quiz-backend/src/Course.cpp
#include "Course.h"
//test
// constructor
Course::Course(const std::string& name, const int& cID) : name_(name), courseID_(cID) {}

// adds topic to course
void Course::addTopic(Topic& t) {
    topics_.push_back(t);
}

// returns the index of a given topic. can find topic names thru listTopics()
// probably not for user use
int Course::findTopic(const int& tID) const {
    for (auto it = topics_.begin(); it != topics_.end(); it++) {
        if (it->getTopicID() == tID) {
            return std::distance(topics_.begin(), it);
        }
    }
    return -1; // topic not found
}

// precondition: topics_ is not empty
// probably not for user use
void Course::listTopics() const {
    std::cout << "Topics currently in " << courseName_ << ": ";
    for (unsigned int i = 0; i < topics_.size() - 1; i++) {
        std::cout << topics_[i].getName() << std::endl;
        std::cout << ", " << std::endl;
    }
    std::cout << topics_[topics.size() - 1].getName() << std::endl;
}

std::vector<Topic> Course::getTopics() {
    return topics_;
}

int Course::getCourseID() const {
    return courseID_;
}

File: quiz-backend/src/models/Course.py
class Course:
    def __init__(self, name, c_id):
        self.course_name = name
        self.course_id = c_id
        self.topics = []

    def add_topic(self, topic):
        self.topics.append(topic)

    def find_topic(self, t_id):
        for index, topic in enumerate(self.topics):
            if topic.get_topic_id() == t_id:
                return index
        return -1  

    def list_topics(self):
        print(f"Topics currently in {self.course_name}:")
        for topic in self.topics:
            print(f"- {topic.get_name()}")

    def get_topics(self):
        return self.topics

    def get_course_id(self):
        return self.course_id
    

    